src/theory/strings/regexp_entail.cpp

```diff
@@ -696,6 +703,70 @@ Node RegExpEntail::getFixedLengthForRegexp(Node n)
   return Node::null();
 }
 
+Node RegExpEntail::getConstantBoundLengthForRegexp(TNode n, bool isLower) const
+{
+  Assert(n.getType().isRegExp());
+  Node ret = getConstantBoundCache(n, isLower);
+  if (!ret.isNull())
+  {
+    return ret;
+  }
+  Kind k = n.getKind();
+  NodeManager* nm = NodeManager::currentNM();
+  if (k == STRING_TO_REGEXP)
+  {
+    ret = d_aent.getConstantBoundLength(n[0], isLower);
+  }
+  else if (k == REGEXP_ALLCHAR || k == REGEXP_RANGE)
+  {
+    ret = nm->mkConst(Rational(1));
+  }
+  else if (k == REGEXP_UNION || k == REGEXP_INTER || k == REGEXP_CONCAT)
+  {
+    bool success = true;
+    bool firstTime = true;
+    Rational rr(0);
+    for (const Node& nc : n)
+    {
+      Node bc = getConstantBoundLengthForRegexp(nc, isLower);
+      if (bc.isNull())
```

Comment:
That seems overly conservative for lower bounds on concats or bounds on intersections.

I think that it should be something like `bc.isNull() && (k == REGEXP_UNION || (k == REGEXP_CONCAT && !isLower))` (and `success` would have to be set differently).

Reply:
Yeah you're right, I'll improve.